app/assets/javascripts/discourse/controllers/composer.js.es6

```diff
@@ -709,6 +709,18 @@ export default Controller.extend({
         if (result.responseJson.action === "create_post") {
           this.appEvents.trigger("post:highlight", result.payload.post_number);
         }
+
+        if (result.responseJson.route_to) {
+          return this.destroyDraft().then(() => {
```

Comment:
I wonder if we should instead show the message first before destroying the draft?

Reply:
The message isn't a confirmation, but what I can do is delete the draft in parallel. I've created a follow up commit, thanks!